openlibrary/core/models.py

```diff
@@ -227,6 +228,38 @@ def _get_d(self) -> dict:
             "l": self._get_lists_cached(),
         }
 
+    def _get_lists_solr(self, limit: int = 50, offset: int = 0, sort: bool = True):
+        # Fetches and caches the lists through Solr, rather than through the DB.
+        # Caching the default case
+        from openlibrary.core.lists.model import List
+
+        if limit == 50 and offset == 0:
+            docs = self._get_lists_solr_cached()
+        else:
+            docs = self._get_lists_solr_uncached(limit=limit, offset=offset)
+        lists = [List(key=doc["key"], site=web.ctx.site) for doc in docs]
```

Comment:
I haven't seen the `Thing` constructors called directly like this before! To be consistent with other spots, let's do


```suggestion
        lists = cast(list[List], web.ctx.site.get_many([doc["key"] for doc in docs]))
```

Note in a future PR we could potentially avoid hitting the DB at all! But this is good for now.